src/Http/Http.Extensions/src/HttpJsonOptionsSetup.cs

```diff
@@ -0,0 +1,24 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json.Serialization.Metadata;
+using Microsoft.AspNetCore.Http.Json;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Http;
+
+public sealed class HttpJsonOptionsSetup : IConfigureOptions<JsonOptions>
```

Comment:
This should be:
```suggestion
public sealed class HttpJsonOptionsSetup : IPostConfigureOptions<JsonOptions>
```

Reply:
@davidfowl I changed to set the DefaultJsonTypeResolver directly when the JsonOptions is created. Are you ok with that?

Reply:
I just changed the behavior after all discussion, and it will throw when we call `JsonOptions.MakeReadOnly()`.